pkg/obfuscate/obfuscate.go

```diff
@@ -0,0 +1,224 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package obfuscate implements quantizing and obfuscating of tags and resources for
+// a set of spans matching a certain criteria.
```

Comment:
It would be good to add here something similar to this https://github.com/DataDog/datadog-agent/blob/8a8e51489c3659de0069489c480534f9dd4d5956/pkg/otlp/model/doc.go#L9-L12 documenting
1. where the module will be used,
2. that its public API is unstable and we provide no guarantees

Reply:
Yes. I am working exactly on that now. Looking at `godoc` locally and seeing what needs to go and how things can be improved.